### DISCUSSION TITLE
Created new collection but Pocketbase says the table does not exist when I create a new record

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#4377] [Q&A] [answered]
Hi,
the title says everything. So i created this collection "priceAlert" but when I want to create a new record it gives me an error 400 status.

![Screenshot from 2024-02-19 02-57-33](https://github.com/pocketbase/pocketbase/assets/22890183/e6ede33e-5282-4e36-85d5-b2a6e2cbdad2)

![Screenshot from 2024-02-19 02-56-37](https://github.com/pocketbase/pocketbase/assets/22890183/e4d621c8-bfbf-4a6d-b2e8-5d740965c5c2)

What I did is to copy the pb_data and pb_migrations from my local computer to my server

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. Make sure that you are testing on the correct environment.

**Reply by ganigeorgiev** [reply]
Also I remember you from https://github.com/pocketbase/pocketbase/discussions/3867.

If you are still using Marmot (or other SQLite replication tool) make sure that the underlying tables were copied (as mentioned in the discussion, Marmot doesn't propagate schema changes).

**Reply by ganigeorgiev** [reply]
Also coming from https://github.com/pocketbase/pocketbase/discussions/3876, if you use `Save(collection)` instead of `SaveCollection(collection)` in your migrations then this will save only the collection model and will not create/update the underlying records table.

**Reply by MuslemRahimi** [reply]
Still not working :(

```
ERROR GET /api/collections/r5n8ziko2uru7of/records?page=1&perPage=40&sort=-created&skipTotal=1&filter=&expand=user&fields=*,expand.user.*:excerpt(200)
└─ Something went wrong while processing your request.
   └─ SQL logic error: no such table: priceAlert (1); failed query: SELECT `priceAlert`.* FROM `priceAlert` ORDER BY [[priceAlert.created]] DESC LIMIT 40
ERROR GET /api/collections/r5n8ziko2uru7of/records?page=1&perPage=1&filter=&fields=id
└─ Something went wrong while processing your request.
   └─ SQL logic error: no such table: priceAlert (1); failed query: SELECT `priceAlert`.* FROM `priceAlert` LIMIT 1

```

this is my pb_migrations file
```
/// <reference path="../pb_data/types.d.ts" />
migrate((db) => {
  const dao = new Dao(db)
  const collection = dao.findCollectionByNameOrId("r0qqxf6uk0sycwd")

  // remove
  collection.schema.removeField("vlr74ews")

  // add
  collection.schema.addField(new SchemaField({
    "system": false,
    "id": "0ogzuivb",
    "name": "priceAlert",
    "type": "relation",
    "required": false,
    "presentable": false,
    "unique": false,
    "options": {
      "collectionId": "r5n8ziko2uru7of",
      "cascadeDelete": true,
      "minSelect": null,
      "maxSelect": 1,
      "displayFields": null
    }
  }))

  return dao.saveCollection(collection)
}, (db) => {
  const dao = new Dao(db)
  const collection = dao.findCollectionByNameOrId("r0qqxf6uk0sycwd")

  // add
  collection.schema.addField(new SchemaField({
    "system": false,
    "id": "vlr74ews",
    "name": "priceAlert",
    "type": "relation",
    "required": false,
    "presentable": false,
    "unique": false,
    "options": {
      "collectionId": "r5n8ziko2uru7of",
      "cascadeDelete": true,
      "minSelect": null,
      "maxSelect": 1,
      "displayFields": null
    }
  }))

  // remove
  collection.schema.removeField("0ogzuivb")

  return dao.saveCollection(collection)
})

```
I tried it with "save(...) and saveCollection(...)" still not working

**Reply by ganigeorgiev** [reply]
> I tried it with "save(...) and saveCollection(...)" still not working

Note that migration once applied it won't execute second time, so if you previously used `save` instead of `saveCollection` then it is expected to not work. You'll have to run `./pocketbase migrate down [N]` and then `./pocketbase migrate up` to rerun them (see [JS Migrations](https://pocketbase.io/docs/js-migrations/#creating-migrations)).

Another thing to note is that the above migration doesn't create the `priceAlert` table that fails in your error message.

But in all cases this is not really a PocketBase issue as it can't occur in normal operations and most likely it is caused by incorrectly synced Marmot instances. The easiest way to fix that is to wipe the Marmot instances and recreate them from the primary/working one.

**Reply by ganigeorgiev** [reply, marked as answer]
I forgot also to mention that another easy way to fix that could be to create a new snapshot migration from the working instance (`./pocketbase migrate collections`) OR manually export your collection configurations from the _Admin UI > Settings > Export collections_ and then import it in the broken instance from _Admin UI > Settings > Import collections_ but note that the manual import will not produce a migration and your instances could get out of sync again.

**Reply by MuslemRahimi** [reply]
I needed that, thanks!
